src/main/scala/com/vngrs/etl/Transform.scala

```diff
@@ -0,0 +1,64 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * Transforms data ([[org.apache.spark.rdd.RDD]]) to another form.
+  *
+  * @tparam A Type of the data ([[org.apache.spark.rdd.RDD]]).
+  */
+trait Transform[A, B] {
+
+  /**
+    * Runs transformation operation,
+    *
+    * @param input [[org.apache.spark.rdd.RDD]]
+    * @return Transformed [[org.apache.spark.rdd.RDD]]
+    */
+  def apply(input: RDD[A]): RDD[B]
+}
+
+/**
+  * Companion object which acts as a Factory.
+  */
+object Transform {
```

Comment:
When I try to do that, compiler throws an exception;

```
Error:(46, 7) double definition:
method apply:[A, B](f: org.apache.spark.rdd.RDD[A] => org.apache.spark.rdd.RDD[B])(implicit evidence$3: scala.reflect.ClassTag[A], implicit evidence$4: scala.reflect.ClassTag[B])com.vngrs.etl.Transform[A,B] and
method apply:[A, B](f: A => B)(implicit evidence$1: scala.reflect.ClassTag[A], implicit evidence$2: scala.reflect.ClassTag[B])com.vngrs.etl.Transform[A,B] at line 36
have same type after erasure: (f: Function1, implicit evidence$3: reflect.ClassTag, implicit evidence$4: reflect.ClassTag)com.vngrs.etl.Transform
  def apply[A: ClassTag, B: ClassTag](f: RDD[A] => RDD[B]): Transform[A, B] = new Transform[A, B] 
```

However, I can change current `apply` method to `map` so it will not throw an exception and will provide better information what method really does. What about that ?